[SQL] cannot find function `extract_hour_TimeN`, `extract_minute_TimeN`, `extract_second_TimeN`

Following SQL:
```
CREATE TABLE illegal_tbl(
tme TIME
);

CREATE MATERIALIZED VIEW v AS SELECT
HOUR(tme) AS hr,
MINUTE(tme) AS min,
SECOND(tme) AS sec
FROM illegal_tbl;
```
Given example input:
**"tme": "14:23:44"**

Produces the following errors:
```
 error[E0425]: cannot find function `extract_hour_TimeN` in this scope
    --> crates/feldera_pipe_operator_59bd8aa34c4151c3/src/lib.rs:88:17
     |
88   |                 extract_hour_TimeN(p1.3.as_ref().cloned()), 
     |                 ^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `extract_hour_Time`
     |
    ::: /Users/rivuadhikari/feldera/crates/sqllib/src/timestamp.rs:1702:1
     |
1702 | pub fn extract_hour_Time(value: Time) -> i64 {
     | -------------------------------------------- similarly named function `extract_hour_Time` defined here
```
```
 error[E0425]: cannot find function `extract_minute_TimeN` in this scope
    --> crates/feldera_pipe_operator_59bd8aa34c4151c3/src/lib.rs:91:17
     |
91   |                 extract_minute_TimeN(p1.3.as_ref().cloned()), 
     |                 ^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `extract_minute_Time`
     |
    ::: /Users/rivuadhikari/feldera/crates/sqllib/src/timestamp.rs:1696:1
     |
1696 | pub fn extract_minute_Time(value: Time) -> i64 {
     | ---------------------------------------------- similarly named function `extract_minute_Time` defined here
```
```
 error[E0425]: cannot find function `extract_second_TimeN` in this scope
    --> crates/feldera_pipe_operator_59bd8aa34c4151c3/src/lib.rs:94:17
     |
94   |                 extract_second_TimeN(p1.3.as_ref().cloned()));
     |                 ^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `extract_second_Time`
     |
    ::: /Users/rivuadhikari/feldera/crates/sqllib/src/timestamp.rs:1690:1
     |
1690 | pub fn extract_second_Time(value: Time) -> i64 {
     | ---------------------------------------------- similarly named function `extract_second_Time` defined here
```

--- a/crates/sqllib/src/timestamp.rs
+++ b/crates/sqllib/src/timestamp.rs
@@ -2372,6 +2372,12 @@ pub fn extract_hour_Time(value: Time) -> i64 {
     time.hour().into()
 }
 
+some_polymorphic_function1!(extract_millisecond, Time, Time, i64);
+some_polymorphic_function1!(extract_microsecond, Time, Time, i64);
+some_polymorphic_function1!(extract_second, Time, Time, i64);
+some_polymorphic_function1!(extract_minute, Time, Time, i64);
+some_polymorphic_function1!(extract_hour, Time, Time, i64);
+
 #[doc(hidden)]
 pub fn parse_time__(format: SqlString, st: SqlString) -> Option<Time> {
     let nt = NaiveTime::parse_from_str(st.str(), format.str());

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
@@ -794,4 +794,11 @@ public class Regression1Tests extends SqlIoTest {
         ccs.visit(cloneCounter.getCircuitVisitor(false));
         Assert.assertEquals(7, cloneCount[0]);
     }
+
+    @Test
+    public void issue4585() {
+        this.getCCS("""
+                CREATE TABLE T(x TIME);
+                CREATE VIEW V AS SELECT HOUR(x) FROM T;""");
+    }
 }

